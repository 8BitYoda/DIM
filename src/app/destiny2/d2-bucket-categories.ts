import type { DimBucketCategory, DimBucketType } from 'app/inventory/inventory-buckets';

export const D2Categories: {
  [key in DimBucketCategory]: DimBucketType[];
} = {
  Postmaster: ['Engrams', 'LostItems', 'Messages', 'SpecialOrders'],
  Weapons: ['KineticSlot', 'Energy', 'Power'],
  Armor: ['Helmet', 'Gauntlets', 'Chest', 'Leg', 'ClassItem'],
  General: [
    'Class',
    'Ghost',
    'Emblems',
    'Ships',
    'Vehicle',
    'Finishers',
    'SeasonalArtifacts',
    'ClanBanners',
  ],
  Inventory: ['Consumables', 'Modifications'],
};
