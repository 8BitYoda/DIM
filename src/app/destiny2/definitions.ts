import { D1ManifestDefinitions } from '../destiny1/d1-definitions';
import { D2ManifestDefinitions } from '../destiny2/d2-definitions';
export interface ManifestDefinitions {
  /** Check if these defs are from D1. Inside an if statement, these defs will be narrowed to type D1ManifestDefinitions. */
  isDestiny1(): this is D1ManifestDefinitions;
  /** Check if these defs are from D2. Inside an if statement, these defs will be narrowed to type D2ManifestDefinitions. */
  isDestiny2(): this is D2ManifestDefinitions;
}

export class HashLookupFailure extends Error {
  table: string;
  id: number;

  /** Pass in just a message key to set the message to the localized version of that key, or override with the second parameter. */
  constructor(table: string, id: number) {
    super(`hashLookupFailure: ${table}[${id}]`);
    this.table = table;
    this.id = id;
    this.name = 'HashLookupFailure';
  }
}
