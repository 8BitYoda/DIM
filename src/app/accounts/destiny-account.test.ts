import { DestinyLinkedProfilesResponse } from 'bungie-api-ts/destiny2';
import 'cross-fetch/polyfill';
import aceProfile from '../../testing/data/ace-accounts.json';
import vidboiProfile from '../../testing/data/vidboi-accounts.json';
import { DestinyAccount, generatePlatforms } from './destiny-account';

async function getCharactersAPI(_account: DestinyAccount) {
  return [{ id: '1231541', dateLastPlayed: new Date(0), base: '' }];
}

const cases = [
  ['ace', aceProfile],
  ['vidboi', vidboiProfile],
];

describe('generatePlatforms', () => {
  test.each(cases)("%s's profile", async (_name, profile) => {
    const platforms = await generatePlatforms(
      profile as unknown as DestinyLinkedProfilesResponse,
      getCharactersAPI
    );
    expect(platforms).toMatchSnapshot();
  });
});
