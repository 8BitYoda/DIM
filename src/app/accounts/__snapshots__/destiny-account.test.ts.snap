// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePlatforms ace's profile 1`] = `
Array [
  Object {
    "destinyVersion": 2,
    "displayName": "Ace#2060",
    "lastPlayed": 2021-10-11T14:59:45.000Z,
    "membershipId": "4611686018467367772",
    "originalPlatformType": 3,
    "platformLabel": "Steam",
    "platforms": Array [
      2,
      5,
      3,
    ],
  },
  Object {
    "destinyVersion": 1,
    "displayName": "Ace#2060",
    "lastPlayed": 2021-10-12T05:46:14.886Z,
    "membershipId": "4611686018467367772",
    "originalPlatformType": 3,
    "platformLabel": "Steam",
    "platforms": Array [
      2,
    ],
  },
]
`;

exports[`generatePlatforms vidboi's profile 1`] = `
Array [
  Object {
    "destinyVersion": 2,
    "displayName": "VidBoi-BMC#9226",
    "lastPlayed": 2021-10-11T01:46:02.000Z,
    "membershipId": "4611686018433092312",
    "originalPlatformType": 2,
    "platformLabel": "PlayStation",
    "platforms": Array [
      1,
      3,
      5,
      2,
    ],
  },
  Object {
    "destinyVersion": 1,
    "displayName": "VidBoi-BMC#9226",
    "lastPlayed": 2021-10-12T05:46:14.893Z,
    "membershipId": "4611686018433092312",
    "originalPlatformType": 2,
    "platformLabel": "PlayStation",
    "platforms": Array [
      2,
    ],
  },
  Object {
    "destinyVersion": 1,
    "displayName": "VidBoi-BMC#9226",
    "lastPlayed": 2021-10-12T05:46:14.893Z,
    "membershipId": "4611686018429726245",
    "originalPlatformType": 1,
    "platformLabel": "Xbox",
    "platforms": Array [
      1,
    ],
  },
]
`;
